src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -333,13 +333,12 @@ private static int VerifyClientCertificate(int preverify_ok, IntPtr x509_ctx_ptr
         private static unsafe int AlpnServerSelectCallback(IntPtr ssl, out IntPtr outp, out byte outlen, IntPtr inp, uint inlen, IntPtr arg)
         {
             GCHandle protocols = GCHandle.FromIntPtr(arg);
+            Debug.Assert(protocols.IsAllocated && protocols.Target != null);
+                       
             byte[] server = (byte[])protocols.Target;
 
-            fixed (byte* sp = server)
-            {
-                return Interop.Ssl.SslSelectNextProto(out outp, out outlen, (IntPtr)sp, (uint)server.Length, inp, inlen) == Interop.Ssl.OPENSSL_NPN_NEGOTIATED ?
+            return Interop.Ssl.SslSelectNextProto(out outp, out outlen, (IntPtr)protocols.AddrOfPinnedObject(), (uint)server.Length, inp, inlen) == Interop.Ssl.OPENSSL_NPN_NEGOTIATED ?
```

Comment:
I may be confusing it with another API, but doesn't `AddrOfPinnedObject` return an `IntPtr`?  I'm wondering what the cast is for.

Reply:
@Drawaes, the cast is back?  I'd thought you'd removed it, but it appears to have been reverted?

Reply:
It is my bad, I was resetting/dropping test changes to figure out what was going on and I killed that as I had merged locally that change into a cleanup commit. I am reverting that.